### PR TITLE
ONEM-24154 Wait for SetURL result (with timeout). Moved workarounds code into

### PR DESCRIPTION
**Pull request by wlodzimierz-lipert-red**
separate function.

### CODE REVIEW

File: WebKitBrowser/WebKitImplementation.cpp [hunk 1]
```diff
@@ -2072,10 +2153,13 @@ static GSourceFuncs _handlerIntervention =
                 return (Core::ERROR_INCOMPLETE_CONFIG);
             }
 
-            bool environmentOverride(WebKitBrowser::EnvironmentOverride(_config.EnvironmentOverride.Value()));
+            const bool environmentOverride(WebKitBrowser::EnvironmentOverride(_config.EnvironmentOverride.Value()));
 
-            if ((environmentOverride == false) || (Core::SystemInfo::GetEnvironment(_T("WPE_WEBKIT_URL"), _URL) == false)) {
-                _URL = _config.URL.Value();
+            {
```
Comment (maciej-kolanski-red) [inline WebKitBrowser/WebKitImplementation.cpp:2158]: What's the point of this inner-scope?

File: WebKitBrowser/WebKitImplementation.cpp [hunk 2]
```diff
@@ -1362,75 +1397,89 @@ static GSourceFuncs _handlerIntervention =
...
+        {
+            std::unique_lock<std::mutex> lock{urlData_.mutex};
+            return urlData_.url;
+        }
+
+        std::string urlValue(std::string url)
+        {
+            std::unique_lock<std::mutex> lock{urlData_.mutex};
+            std::swap(urlData_.url, url);
+            return url;
```
Comment (maciej-kolanski-red) [inline WebKitBrowser/WebKitImplementation.cpp:1410]: Returned value is never used, but it's still surprising that setter returns the old member's value.
Consider returning urlData_.url or void.

**Review by maciej-kolanski-red — COMMENTED** [review]
Fix commit message (add ticket ID and short title)

File: WebKitBrowser/WebKitImplementation.cpp (hunk 2 above)
Comment (wlodzimierz-lipert-red) [reply WebKitBrowser/WebKitImplementation.cpp:1410]: its setter function.

**Review by maciej-kolanski-red — APPROVED** [review]

**Pull request merged**